wrapper/src/main/java/software/amazon/jdbc/ConnectionInvalidator.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package software.amazon.jdbc;
+
+import java.lang.ref.WeakReference;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.Map;
+import java.util.Queue;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+public class ConnectionInvalidator {
+
+  public static final Map<String, Queue<WeakReference<Connection>>> OPENED_CONNECTION =
+      new ConcurrentHashMap<>();
+  private static final ExecutorService invalidateConnectionsExecutorService =
+      Executors.newCachedThreadPool(
+          r -> {
+            final Thread invalidateThread = new Thread(r);
+            invalidateThread.setDaemon(true);
+            return invalidateThread;
+          });
+
+  /**
+   * Invalidates all opened connections pointing to the same node in a daemon thread.
+   *
+   * @param hostSpec The {@link HostSpec} object containing the url of the node.
+   */
+  public static void invalidateAllConnections(final HostSpec hostSpec) {
+    invalidateConnectionsExecutorService.submit(() -> {
+      final Queue<WeakReference<Connection>> connectionQueue = OPENED_CONNECTION.get(
+          hostSpec.getUrl());
+
+      WeakReference<Connection> connReference;
+      while ((connReference = connectionQueue.poll()) != null) {
+        final Connection conn = connReference.get();
+        if (conn == null) {
+          continue;
+        }
+
+        // Try to call rollback in case the connection is in
+        try {
+          conn.rollback();
```

Comment:
Hmm I'm not sure we need it. Calling rollback() might trigger failover that takes time. After failover is completed we close connection.

Reply:
if this is in a pool, but not open in the application then close() would have been called. 
We should be able to figure out if the connection is in the app or in the pool.